core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -492,12 +493,24 @@ public void failTaskRemotely(TaskId taskId, Throwable failureCause)
             SqlStage sqlStage = requireNonNull(stages.get(taskId.getStageId()), () -> "stage not found: %s" + taskId.getStageId());
             sqlStage.failTaskRemotely(taskId, failureCause);
         }
+
+        public void logDebugInfo()
+        {
+            if (!log.isDebugEnabled()) {
+                return;
+            }
+            log.debug("SqlStages:");
+            stages.forEach((stageId, stage) -> {
+                log.debug("SqlStage %s: %s", stageId, stage.getDebugInfo());
```

Comment:
These lines may become separated.
Maybe construct a longer log message (including new lines) and log it once?

Reply:
I will keep it that way. I want to be on the safe side when it comes to logging framework payload size constraints.